.github/workflows/golangci-lint.yml

```diff
@@ -2,8 +2,6 @@ name: golangci-lint
 on:
   pull_request:
     types: [opened, edited, synchronize, reopened]
-    branches:
-      - main
```

Comment:
Q: from which branch the linter config will be taken? from main or from the target branch?
Q: If I got this right, we are going to run a newer version of lint on old branches (previously checked with older versions). Could this introduce some problem?

Reply:
First Q. I think ~ target branch:
> The .github/workflows directory in your repository is searched for workflow files at the associated commit SHA or Git ref. The workflow files must be present in that commit SHA or Git ref to be considered.
> 
> For example, if the event occurred on a particular repository branch, then the workflow files must be present in the repository on that branch.
> https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#example-using-multiple-events-with-activity-types-or-configuration

Reply:
What @sbueringer said is correct. We would still be running the same version of lint that's checked in in the old branches. In order to validate that we can look at the golangci-lint job that will run on the backports of this PR on the release branch.